packages/opencv/lib/index.js

```diff
@@ -344,57 +294,42 @@ async function getImagesMatches(img1Data, img2Data, options = {}) {
       img1.delete();
       img2.delete();
       detector.delete();
-      result1.keyPoints.delete();
-      result1.descriptor.delete();
-      result2.keyPoints.delete();
-      result2.descriptor.delete();
+      result1?.keyPoints.delete();
+      result1?.descriptor.delete();
+      result2?.keyPoints.delete();
+      result2?.descriptor.delete();
       matcher.delete();
       matchesVec.delete();
     } catch (ign) {}
   }
 }
 
 /**
- * @typedef SimilarityOptions
- * @property {?boolean} visualize [false] Whether to return the resulting visalization
- * as an image (useful for debugging purposes)
- * @property {string} method [TM_CCOEFF_NORMED] The name of the template matching method.
- * Acceptable values are:
- * - TM_CCOEFF
- * - TM_CCOEFF_NORMED (default)
- * - TM_CCORR
- * - TM_CCORR_NORMED
- * - TM_SQDIFF
- * - TM_SQDIFF_NORMED
- * Read https://docs.opencv.org/3.0-beta/doc/py_tutorials/py_imgproc/py_template_matching/py_template_matching.html
- * for more details.
- */
-
-/**
- * @typedef SimilarityResult
- * @property {number} score The similarity score as a float number in range [0.0, 1.0].
- * 1.0 is the highest score (means both images are totally equal).
- * @property {?Buffer} visualization The visualization of the matching result
- * represented as PNG image buffer. This image includes both input pictures where
- * difference regions are highlighted with rectangles.
+ * @param {Buffer} buf
+ * @returns {string}
  */
+function base64Encode(buf) {
```

Comment:
not sure we need a dedicated function for that

Reply:
we use it in a couple places, but it makes sense organizationally as well